_search-plugins/ubi/schemas.md

```diff
@@ -0,0 +1,230 @@
+- The **Search client**, if separate from the **UBI client**, forwards the indexed [`query_id`](#query_id) to the **UBI client**.
+    Even though the the roles of *search* and *UBI event indexing* are separated on this diagram, many implementations can use the same OpenSearch client instance for both roles of searching and index writing. 
+    <br/>(6 in the preceding diagram).
+    {: .note} 
+- The **UBI client** then indexes all user events with the specified [`query_id`](#query_id) until a new search is performed. At this time, a new `query_id` is generated by **User Behavior Insights** plugin and passed back to the **UBI client**.
+- If the **UBI client** interacts with a result **object**, such as during an add to cart event, then the [`object_id`](#object_id), `add_to_cart` [`action_name`](#action_name) and `query_id` are all indexed together, signaling the causal link between the *search* and the *object*.
+ <br/>(8 and 9, preceding diagram)
+
+
+
+## UBI stores
+
+There are two separate stores involved in supporting UBI data collection:
+* UBI queries
+* UBI events
+
+### UBI queries index
+
+All underlying query information and results ([object IDs](#object_id)) are stored in the `ubi_queries` index, and remain largely invisible in the background.
+
+
+The `ubi_queries` index [schema](https://github.com/OpenSearch-project/user-behavior-insights/tree/main/src/main/resources/queries-mapping.json) has the following fields:
+
+- `timestamp` (events and queries): A UNIX timestamp of when the query was received.
+
+- `query_id` (events and queries): A unique ID of the query provided by the client or generated automatically. Different queries with the same text generate different `query_id` values. 
+	
+- `client_id` (events and queries): A user/client ID provided by the client application.
+
+- `query_response_objects_ids` (queries):  An array of [object IDs](#object_id). An ID can have the same value as the `_id` but it is meant to be the externally valid ID of a document, item, or product.
+
+Since UBI manages the `ubi_queries` index, you should never have to write directly to this index (except for importing data).
+
+### UBI events index
+
+The client side directly indexes events to the `ubi_events` index, linking the event [`action_name`](#action_name), objects (each with an [`object_id`](#object_id)), and queries (each with a [`query_id`](#query_id)) along with any other important event information.
+Because this schema is dynamic, you can add any new fields and structures (such as user information or geo-location information) that are not in the current **UBI Events** [schema](https://github.com/opensearch-project/user-behavior-insights/tree/main/src/main/resources/events-mapping.json) at index time.
```

Comment:
```suggestion
Because this schema is dynamic, you can add any new fields and structures (such as user information or geo-location information) that are not in the current **UBI events** [schema](https://github.com/opensearch-project/user-behavior-insights/tree/main/src/main/resources/events-mapping.json) at index time.
```